src/guide/configuring-labs.md

```diff
@@ -0,0 +1,117 @@
+
+```yaml
+dockerImageId: docker_image_name
+```
+
+The following docker images are supported (more will be added in the near future):
+
+- **keras_v2-0-x_python_3-1** - Keras 2.0.x and Python 3.1
+- **keras_v2-0-x_python_2-1** - Keras 2.0.x and Python 2.1
+
+This means, if we want to run our lab with Keras version 2.0.x and Python version 3.1, our configuration would look like this:
+
+```yaml
+dockerImageId: keras_v2-0-x_python_3-1
+```
+
+Configuring custom docker images is a feature that is planned for future versions.
+
+## Lab inputs (datasets)
+
+Obviously there's no Machine Learning without data. Labs need access to datasets and one way to get hold of them is using **inputs**. Inputs are basically metadata that describe what data needs to be downloaded before a lab is executed.
+
+An input has a `url`, which is the endpoint from which that data should be downloaded from, and a `name`, which specifies under what name the data is stored on the file system in a special `inputs` directory, so it can be later accessed by the lab. MachineLabs has full internet access so we can basically download from any reachable place.
+
+<p class="tip">All inputs are downloaded in a special `inputs` directory and can be accessed from within our labs using common file system operations.</p>
+
+For example, the following configuration sets up an input that fetches the famous MNIST dataset and saves it as `mnist.npz` into the `inputs `directory.
+
+```yaml
+inputs:
+  - name: mnist.npz
+    url: https://s3.amazonaws.com/img-datasets/mnist.npz
+```
+
+If there's multiple datasets to download, no problem. We can simply add more inputs to the list. All inputs will then be downloaded concurrently:
+
+```yaml
+inputs:
+  - name: mnist.npz
+    url: https://s3.amazonaws.com/img-datasets/mnist.npz
+  - name: reuters.npz
+    url: https://s3.amazonaws.com/text-datasets/reuters.npz
+  - name: imdb.npz
+    url: https://s3.amazonaws.com/text-datasets/imdb.npz
+```
+
+<p class="tip">Soon MachineLabs will support **mounting custom datasets** that can be uploaded and shared across labs, so data doesn't have to be downloaded every single time a lab is executed.</p>
+
+## Script parameters
+
+Every now and then we would like to allow configurable parameters for our own scripts or execute third-party scripts that expect parameters. Script parameters can be configured using the `parameters` property. `parameters` is a list of parameters that will be passed to our entry file (e.g. `main.py`) in the same order they are specified in the `ml.yaml`.
+
+<p class="tip">Script parameters are executed in the order they are specified, allowing positional arguments as well!</p>
+
+Sometimes parameters can be configuration values for a script, sometimes they specify input or output data for a script. That's why `parameters` is a list of `pass-as` properties that gives us all the freedom we need to configure different kind of script parameters.
```

Comment:
>Sometimes parameters can be configuration values for a script, sometimes they specify input or output data for a script

I think the wording is a bit confusing here. Aren't parameters always configuration values for a script? Also

>sometimes they specify input or output data

You mean, you specify a directory like `--input-data=./inputs/foo`, right? But after all, that's just another configuration value. I think it would be better to just say that parameters are passed in the order that they are defined so that you can pass positional parameters as well as flags or any other kind of parameter.

Reply:
Okay, I'll change that.